nautobot/ipam/tests/test_models.py

```diff
@@ -174,17 +182,81 @@ def test_prefix_validation(self):
+        self.assertFalse(self.root.is_child_node())
+        self.assertTrue(self.parent.is_child_node())
+        self.assertTrue(self.child1.is_child_node())
+
+        # is_leaf_node()
+        self.assertFalse(self.root.is_leaf_node())
+        self.assertTrue(self.child1.is_leaf_node())
+
+        # is_root_node()
+        self.assertTrue(self.root.is_root_node())
+        self.assertFalse(self.child1.is_root_node())
 
-        self.assertSetEqual(set(duplicate_prefix_pks), {prefixes[1].pk, prefixes[2].pk})
+        # ancestors()
+        self.assertEqual(list(self.child1.ancestors()), [self.root, self.parent])
+        self.assertEqual(list(self.child1.ancestors(ascending=True)), [self.parent, self.root])
+        self.assertEqual(list(self.child1.ancestors(include_self=True)), [self.root, self.parent, self.child1])
 
-    def test_get_child_prefixes(self):
+        # get_children()
+        self.assertEqual(list(self.parent.get_children()), [self.child1, self.child2])
+
+        # descendants()
+        self.assertEqual(list(self.root.descendants()), [self.parent, self.child1, self.child2])
+        self.assertEqual(
+            list(self.root.descendants(include_self=True)), [self.root, self.parent, self.child1, self.child2]
+        )
+
+        # get_root()
+        self.assertEqual(self.child1.get_root(), self.root)
+        self.assertIsNone(self.root.get_root())
+
+        # get_siblings()
+        self.assertEqual(list(self.child1.get_siblings()), [self.child2])
+        self.assertEqual(list(self.child1.get_siblings(include_self=True)), [self.child1, self.child2])
+        parent2 = Prefix.objects.create(prefix="101.102.0.128/25", status=self.status)
+        self.assertEqual(list(self.parent.get_siblings()), [parent2])
+        self.assertEqual(list(self.parent.get_siblings(include_self=True)), [self.parent, parent2])
+
+    # TODO(jathan): When Namespaces are implemented, these tests must be extended to assert it.
+    def test_reparenting(self):
+        """Test that reparenting algorithm works in its most basic form."""
+        # tree hierarchy
+        self.assertIsNone(self.root.parent)
+        self.assertEqual(self.parent.parent, self.root)
+        self.assertEqual(self.child1.parent, self.parent)
+
+        # Delete the parent (/25); child1/child2 now have root (/24) as their parent.
+        # Parent has children, so delete will fail
+        self.assertRaises(models.ProtectedError, self.parent.delete)
+        # Force delete to coerce reparenting
+        num_deleted, _ = self.parent.delete(force_delete=True)
+        self.assertEqual(num_deleted, 1)
+
+        self.assertEqual(list(self.root.get_children()), [self.child1, self.child2])
+        self.child1.refresh_from_db()
+        self.child2.refresh_from_db()
+        self.assertEqual(self.child1.parent, self.root)
+        self.assertEqual(self.child2.parent, self.root)
+        self.assertEqual(list(self.child1.ancestors()), [self.root])
```

Comment:
Can we add to this test to recreate the /25 and verify that the /26s get reparented under it?